### PR TITLE
Support pure-dictionary input format for OD models

### LINKED ISSUE #1342: Models should support pure-dictionary format
Currently, models expect inputs in the format of
`(images, {"boxes": ..., "labels": ...}`.

They should all support pure-dictionary inputs.

### PR DESCRIPTION
**Pull request by ianstenbit**
Fixes #1342

This also updates our GPU CI to actually run `INTEGRATION` tests, since our model tests are no longer so expensive that they time out the GPU tests. In order to do this, I had to add a separate `REGRESSION` control to two tests which are currently very expensive that don't belong in our CI.

### CODE REVIEW

**Comment by ianstenbit** [comment]
cc @martin-gorner

File: keras_cv/models/object_detection/faster_rcnn.py [hunk 1]
```diff
@@ -487,9 +488,8 @@ def compute_loss(self, images, boxes, classes, training):
         )
 
     def train_step(self, data):
-        images, y, sample_weight = tf.keras.utils.unpack_x_y_sample_weight(data)
-        if sample_weight is not None:
-            raise ValueError("`sample_weight` is currently not supported.")
+        images, y = unpack_input(data)
```
Comment (LukeWood) [inline keras_cv/models/object_detection/faster_rcnn.py:491]: One issue here: sample weight.  I guess we just don’t support it with the dict input?

Comment (ianstenbit) [reply keras_cv/models/object_detection/faster_rcnn.py:491]: Correct. (Although this is not a regression -- we already don't support sample weight in the OD models)

**Review by LukeWood — APPROVED** [review]
